plaza/routes/files/files.go

```diff
@@ -44,10 +45,29 @@ type file_t struct {
 }
 
 func Post(w http.ResponseWriter, r *http.Request) {
+	var dst *os.File
 	username := r.URL.Query()["username"][0]
 	filename := r.URL.Query()["filename"][0]
-
-	dst, err := os.Create(fmt.Sprintf("/home/%s/%s", username, filename))
+	path := fmt.Sprintf("/home/%s/%s", username, filename)
+
+	_, err := os.Stat(path)
+	// if a file with exactly the same name already exists
+	if err == nil {
+		// we rename it like 'file (2).txt'
+		for i := 1; i <= 50; i++ {
```

Comment:
from 1 to 50 ? Why should you limit yours users to 50 file with the same name ?

I think it would be better to make a "while" loop here. Anyway, this thing is limited by file system so the loop will never be endless